lib/guid.ts

```diff
@@ -1,15 +1,21 @@
 export class Guid {
     /** Empty GUID string (hyphenated). */
-    public static EMPTY = "00000000-0000-0000-0000-000000000000";
+    public static EMPTY: string = "00000000-0000-0000-0000-000000000000";
     private static validator: RegExp = new RegExp("^[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}$", "i");
-    
-    public static isValid(guid: any) {
-        const value: string = guid.toString();
-        return guid && (guid instanceof Guid || this.validator.test(value));
+
+    public static isValid(guid: string | Guid): boolean {
+        if (guid == undefined) return false;
```

Comment:
you could do a null check to validate null and undefined (just for safety reasons)

```suggestion
        if (guid == null) return false;
```

Reply:
What about removing the whole line instead of enhancing the `undefined``value check? The function will try to parse any non-string values as valid GUIDs and returns false if there is an exception. I'll add a test to support this thesis.